lua/null-ls/builtins/formatting/nix_flake_fmt.lua

```diff
@@ -256,6 +256,15 @@ end
 local find_nix_fmt = function(opts, done)
     done = vim.schedule_wrap(done)
 
+    -- A malicious project could make `nix fmt` do anything to your computer,
+    -- so we ask the user if the project is trusted before we do that.
+    local is_project_trusted = vim.secure.read(opts.root)
```

Comment:
It's super non-obvious what triggers this message. See screenshot:

![image](https://github.com/user-attachments/assets/036897b6-f4f3-40af-98e1-5a96ef73e871)

Can we add more text to make it clear that this is about none-ls's nix_flake_fmt formatter?

Reply:
Additional idea: only eagerly discover the entrypoint if the project is trusted. If the project is not trusted, wait for the user to format.